secrets/config.go

```diff
@@ -0,0 +1,29 @@
+package secrets
+
+import (
+	"context"
+	"time"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// Config is the confuration struct for the secrets package.
+//
+// Can be deserialized from YAML.
+type Config struct {
+	// Path is the path to the secrets.json file file to load your service's
+	// secrets from.
+	Path string
+}
+
+// InitFromConfig returns a new *secrets.Store using the given context and config.
+func InitFromConfig(ctx context.Context, cfg Config) (*Store, error) {
```

Comment:
:no_bell: `NewFromConfig`?

Reply:
I can see either approach 😅 

1. The nice thing about this is it follows the same convention as the other packages so it's clear what it's for.
2. The part that is a little weird is it's the only one that isn't setting up some global state and just returning an `io.Closer` so I can see why renaming it might make sense.

What does everyone else think?

Reply:
I slightly prefer `InitFromConfig` to keep it consistent.